integration-tests/chopsticks/src/network/utils.ts

```diff
@@ -1,53 +1,73 @@
-export function getSiblingLocation(paraId: number) {
-	return {
-		parents: 1,
-		interior: {
-			X1: { Parachain: paraId },
-		},
-	}
-}
+import { connectParachains, connectVertical } from '@acala-network/chopsticks'
+import { setTimeout } from 'timers/promises'
+import { setupContext, SetupOption } from '@acala-network/chopsticks-testing'
+
+import type { Config } from './types.js'
 
-export function getParentLocation() {
-	return {
-		parents: 1,
-		interior: 'Here',
+/**
+ * This function is used to shut down a network composed of multiple chains.
+ *
+ * @param {Config[]} chains - An array of chain configurations that make up the network.
+ *
+ * @returns {Promise<void>}
+ * Returns a Promise that resolves when all chains in the network have been successfully shut down.
+ *
+ */
+export async function shutDownNetwork(chains: Config[]): Promise<void> {
+	await setTimeout(50)
+	const tearDown = chains.map((chain) => chain.teardown())
+	await Promise.all(tearDown)
+}
+const newBlock = async (newBlockConfig: { count: number }, contexts: Config[]) => {
+	await Promise.all(contexts.map((context) => context.dev.newBlock(newBlockConfig)))
+}
+async function connectNetworks(relayChain: Config, parachains: Config[]) {
+	for (const parachain of parachains) {
+		await connectVertical(relayChain.chain, parachain.chain)
 	}
+
+	await connectParachains(parachains.map((parachain) => parachain.chain))
+
+	const newBlockConfig = { count: 2 }
+	// fixes api runtime disconnect warning
+	await setTimeout(50)
+	// Perform runtime upgrade and establish xcm connections.
+	await newBlock(newBlockConfig, [relayChain, ...parachains])
 }
 
-export function getAccountLocationV2(addr: string) {
-	return {
-		V2: {
-			parents: 0,
-			interior: {
-				X1: {
-					AccountId32: {
-						network: 'Any',
-						id: addr,
-					},
-				},
-			},
-		},
-	}
+/**
+ * This function is used to set up a network with a relay chain, a sender, and a receiver.
+ *
+ * @param {SetupOption} relayChain - The relay chain option for the network setup.
+ * @param {SetupOption} sender - The sender option for the network setup.
+ * @param {SetupOption} receiver - The receiver option for the network setup.
+ *
+ * @returns {Promise<{relayChainContext: Config, senderChainContext: Config, receiverChainContext: Config}>}
+ * An object containing the contexts of the relay chain, sender, and receiver.
+ */
+export async function setupNetwork(
+	relayChain: SetupOption,
+	sender: SetupOption,
```

Comment:
This as well could involve configuring more than a single receiver chain, so it should probably be generalized.

Reply:
If you need to treat the realy chain differently because of a different function name, then I would simply have the relay chain argument and a list of chains to all connect horizontally with each other and vertically with the relaychain. As far as I can see, you don't need to know who the sender is here.